version: "3.7"
services:
  shinobi:
    image: shinobisystems/shinobi:latest-ubuntu
#    user: "2022:2022"
#    ports:
#      - "8080"
    labels:
      - "traefik.http.routers.whoami.rule=Host(`shinobi`)"
      - "traefik.http.services.shinobi.loadbalancer.server.port=8080"
#      - traefik.frontend.rule=Host:shinobi
#      - traefik.port=8080
      - "traefik.docker.network=traefik_default"
    environment:
      - TZ=America/New_York
      - APP_BRANCH=dev
      - DEBUG=1
      - PUID=2022
      - PGID=2022
    networks:
      outside-world:
    volumes:
      - "/dev/shm/shinobiStreams:/dev/shm/streams:Z"
      - "/docker/nfs_mount/shinobi/shinobiConfig:/config:Z"
      - "/docker/nfs_mount/shinobi/shinobiCustomAutoLoad:/customAutoLoad:Z"
      - "/docker/nfs_mount/shinobi/shinobiDatabase:/var/lib/mysql:Z"
      - "/docker/nfs_mount/shinobi/shinobiVideos:/opt/shinobi/videos:Z"

# proxy:
#    dockerfile: nginx.dockerfile
#    container_name: proxy
#    restart: unless-stopped
#    ports:
#      - "8443:8443"
##    volumes:
##      - /etc/docker/compose/shinobi/nginx/:/etc/nginx/conf.d/
##      - web-root:/var/www/html
##      - certbot-etc:/etc/letsencrypt
##      - certbot-var:/var/lib/letsencrypt
#    depends_on:
#      - shinobi
#    networks:
#      - appnet


#networks:
#  appnet:
networks:
  outside-world:
    external:
      name: traefik_outside-world
